wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,462 @@
+		this.source = source
+		this.target = target
+		this.initialAmount = amount
+		this.amount = amount
+		this.ranged = ranged
+		this.attack = attack
+		this.attackType = attackType
+		this.damageType = damageType
+		this.weaponType = weaponType
+		this.data = data
+		this.isReturned = isReturned
+
+	/** Allows manual creation of Damage instance, which is then dealt
+		through .deal() call, guaranteed passing it to modifiers and listeners.
+
+		You can attach anything to Damage instance with .data field.
+		isReturned specifies whether the Damage should not be destroyed during the event. */
+	construct(unit source, unit target, real amount, bool ranged, bool attack, attacktype attackType, damagetype damageType, weapontype weaponType, int data, bool isReturned)
+		set(source, target, amount, ranged, attack, attackType, damageType, weaponType, data, isReturned)
+
+	construct(unit source, unit target, real amount, attacktype attackType, int data, bool isReturned)
+		set(source, target, amount, false, false, attackType, DAMAGE_TYPE_UNIVERSAL, WEAPON_TYPE_WHOKNOWS, data, isReturned)
+	
+	construct(unit source, unit target, real amount, int data, bool isReturned)
+		set(source, target, amount, false, false, ATTACK_TYPE_CHAOS, DAMAGE_TYPE_UNIVERSAL, WEAPON_TYPE_WHOKNOWS, data, isReturned)
+
+	/** applies user-created Damage instance, instantly processing it through
+		all damage events. After this function is called, the Damage will be
+		in 'fired' state. */
+	function deal()
+		if state == DamageState.CREATED
+			// What is described in hotdoc happens in .invoke()
+			// While this is a wrapper to handle recursive damage properly
+			if recursionAmount > RECURSION_TRESHOLD
+				Log.debug("Damage.deal(): Recursion treshold has been hit.")
+				return
+			state = DamageState.QUEUED
+			queue.push(this)
+			if not dequeueRunning
+				if needTimer
+					if nullTimer == null
+						nullTimer = getTimer()..start(0, function dequeue)
+				else
+					dequeue()
+		else
+			Log.debug("Damage.deal() called on already dealt damage!")
+
+	protected function invoke()
+		state = DamageState.FIRING
+		let temp = dealingInstance
+		dealingInstance = this
+		Log.trace("deal d" + I2S(this castTo int))
+		UnitDamageTarget(source, target, amount, attack, ranged, attackType, damageType, weaponType)
+		Log.trace("completed d" + I2S(this castTo int))
+		dealingInstance = temp
+		if state != DamageState.FIRED
+			Log.error("Damage.deal() returned not fully processed damage! (state: " + state.toString() + ")")
+			if not isReturned
+				destroy this
+			
+	// Event modification
+
+	/** Make actual amount leave target at at least 0.406 hp left */
+	function preventDeath()
+		doPreventDeath = true
+
+	/** Nullifies actual amount no matter what. */
+	function deflect()
+		doDeflect = true
+
+	/** Increases `amount` of the damage by `initialAmount * magnitude` */
+	function addMultiplier(real magnitude)
```

Comment:
I'm not a huge fan of these `addEtc` functions. I'd rather just have the `setAmount` function, and let the users implement multipliers or flat adds as they please. These functions aren't saving enough user effort to warrant their existence.

Reply:
I think the funcs can be fine, but they should all just use the `setAmount` function and not be boilerplate.
Also, `Log.debug` is pointless for user feedback. The default level is info, so this just isn't displayed at all 99% of the time.
It should at least be warning level, depending on the func even be `error()` because continuing with the error doesn't make sense.